src/transpose/index.js

```diff
@@ -4,6 +4,126 @@
+        } else if (i % lengthOfInnerArray === 8) {
+            anotherArray9.push(newOuterArray[i]);            
+        } else if (i % lengthOfInnerArray === 9) {
+            anotherArray10.push(newOuterArray[i]);            
+        } else if (i % lengthOfInnerArray === 10) {
+            anotherArray11.push(newOuterArray[i]);            
+        } 
+    }
+    if (lengthOfInnerArray === 1) {
+        newArray.push(anotherArray1);
+    } else if (lengthOfInnerArray === 2) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+    } else if (lengthOfInnerArray === 3) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+    } else if (lengthOfInnerArray === 4) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+    } else if (lengthOfInnerArray === 5) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+        newArray.push(anotherArray5);
+    } else if (lengthOfInnerArray === 6) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+        newArray.push(anotherArray5);
+        newArray.push(anotherArray6);
+    } else if (lengthOfInnerArray === 7) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+        newArray.push(anotherArray5);
+        newArray.push(anotherArray6);
+        newArray.push(anotherArray7);
+    } else if (lengthOfInnerArray === 8) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+        newArray.push(anotherArray5);
+        newArray.push(anotherArray6);
+        newArray.push(anotherArray7);
+        newArray.push(anotherArray8);
+    } else if (lengthOfInnerArray === 9) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+        newArray.push(anotherArray5);
+        newArray.push(anotherArray6);
+        newArray.push(anotherArray7);
+        newArray.push(anotherArray8);
+        newArray.push(anotherArray9);
+    } else if (lengthOfInnerArray === 10) {
+        newArray.push(anotherArray1);
+        newArray.push(anotherArray2);
+        newArray.push(anotherArray3);
+        newArray.push(anotherArray4);
+        newArray.push(anotherArray5);
+        newArray.push(anotherArray6);
+        newArray.push(anotherArray7);
+        newArray.push(anotherArray8);
+        newArray.push(anotherArray9);
+        newArray.push(anotherArray10);
+    } else if (lengthOfInnerArray === 11) {
```

Comment:
This is problematic, you shouldn't hardcode things like this. If the scenarios change, your solution wouldn't work anymore. That's a problem. You should opt for a more generalised solution always.